data_safe_haven/infrastructure/components/composite/virtual_machine.py

```diff
@@ -276,6 +283,17 @@ def __init__(
             tags=child_tags,
         )
 
+        # Add VM to maintenance configuration
+        self.configuration_assignment_resource = maintenance.ConfigurationAssignment(
+            f"{name_underscored}_configurationAssignment",
+            provider_name="Microsoft.Compute",
+            resource_group_name=props.resource_group_name,
+            resource_name_=virtual_machine.name,
+            resource_type="VirtualMachines",
+            location=props.location,
+            maintenance_configuration_id=props.maintenance_configuration_id,
+        )
```

Comment:
```suggestion
            opts=ResourceOptions.merge(
                child_opts,
                ResourceOptions(parent=virtual_machine),
            ),
        )
```

Reply:
What does this actually do?

Reply:
It's mainly syntactic fluff. It tells Pulumi that to apply all the options that this class wants to set on all its children (`child_opts`) and also set the parent of the resource to be the VM.

We don't (currently) pass anything in `child_opts` but we could in future do [something from here](https://www.pulumi.com/docs/concepts/options/) like setting all the resources to `protected`.